### PR TITLE
Update ``python-docs-theme`` to work with Sphinx 5 & 6

### PR DESCRIPTION
**Pull request by AA-Turner**
Updated remaining jQuery scripts to standard JavaScript, and a few fixes for CSS changes.

A

### CODE REVIEW

**Comment by AA-Turner** [comment]
Turns out this is required for Sphinx 5... https://github.com/python/cpython/pull/99381

A

**Review by pradyunsg — APPROVED** [review]
LGTM

File: python_docs_theme/static/copybutton.js [hunk 1]
```diff
@@ -1,64 +1,63 @@
...
+            codeEl.querySelectorAll('.go, .gp, .gt').forEach(el => el.hidden = true)
+            codeEl.querySelectorAll('.gt').forEach(el => {
+                while ((el = el.nextSibling) && el.nodeType !== Node.DOCUMENT_NODE) {
+                    if (el.nodeType === Node.ELEMENT_NODE && el.matches(".gp, .go")) break;
+                    if (el.nodeType === Node.TEXT_NODE && el.textContent.trim()) {
+                        const wrapper = document.createElement('span');
+                        el.after(wrapper);
+                        wrapper.appendChild(el);
+                        el = wrapper
+                    }
```
Comment (ezio-melotti) [inline python_docs_theme/static/copybutton.js:30]: Why is this necessary?

File: python_docs_theme/static/copybutton.js [hunk 2]
```diff
@@ -1,64 +1,63 @@
...
+            buttonEl.title = show_text
+            buttonEl.dataset.hidden = "true"
         } else {
             // show the code output
-            button.parent().find('.go, .gp, .gt').show();
-            button.next('pre').find('.gt').nextUntil('.gp, .go').css('visibility', 'visible');
-            button.css('text-decoration', 'none');
-            button.attr('title', hide_text);
-            button.data('hidden', 'false');
+            codeEl.childNodes.forEach(el => el.hidden = false)
```
Comment (ezio-melotti) [inline python_docs_theme/static/copybutton.js:38]: IIUC here you are just assuming that everything inside the code block was already visible by default, and you are setting `.hidden = false` to each element.  Do you think this might conflict with other things (e.g. Sphinx extensions) that might introduce hidden elements?

**Review by ezio-melotti — COMMENTED** [review]
In addition to the inline comments, I have two generic comments:
* The use of semicolons at the end of the line seems inconsistent.  Unless there is a specific reason to keep them, I would just remove them all.
* I would avoid the use of `if (...) return;` or `if (...) break;` or even `if (...) some_functions()`.  The expression after the condition should go on a new line, and it should be surrounded by `{ }`, e.g.:
   ```js
   if (...) {
      ...
   }
   ```

---

Fun fact: I originally implemented both the collapsible sidebar (~12 years ago in https://github.com/python/cpython/issues/47393) and the copybutton (~11 years ago in https://github.com/python/cpython/commit/59dba38166160163ce4ac473933ab1c74b6222de) and since then they have traveled around a number of dirs and repos.  It's nice to see that in the meanwhile things got standardized and JS evolved to the point that JQuery is no longer needed.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -1,35 +1,39 @@
----
-
 name: Tests
```
[the remark this answers is not in the document]
Comment (AA-Turner) [reply .github/workflows/tests.yml:1]: This branch is based on top of #94, which is required for CI to pass as of the change to required dependencies -- hopefully #94 is merged soon and I can rebase.

A

File: python_docs_theme/static/copybutton.js (hunk 2 above)
Comment (AA-Turner) [reply python_docs_theme/static/copybutton.js:38]: Yes, this replicates the current behaviour -- I would suggest merging this as-is to keep the behaviour consistent and if the unconditional hiding is a problem we can change it later.

A

File: python_docs_theme/static/copybutton.js (hunk 1 above)
Comment (AA-Turner) [reply python_docs_theme/static/copybutton.js]: I've added a note -- in efect tracebacks have bare text elements that we need to be able to hide, so we wrap in a span. This replicates the current behaviour.

A

File: python_docs_theme/static/copybutton.js (hunk 2 above)
Comment (ezio-melotti) [reply python_docs_theme/static/copybutton.js:42]: The original implementation was using `button.parent().find('.go, .gp, .gt')` `button.next('pre').find('.gt').nextUntil('.gp, .go')` to select the elements, but in this version it seems you are doing this only in the `if` branch but not in the `else`.

Also, without JQuery, selecting the tracebacks seems more convoluted, so I understand you might not want to duplicate it in the `else`, but perhaps you could create a function to show/hide tracebacks and call it in both the `if` and the `else`?  This will make the implementation closer to the original and also more readable/understandable.

Comment (AA-Turner) [reply python_docs_theme/static/copybutton.js]: I've added a new `getHideableCopyButtonElements` generator function.

A

**Comment by hugovk** [comment]
Is this ready to merge?

**Comment by hugovk** [comment]
Rebased on `main` to build RTD preview: 

* https://python-docs-theme-previews--99.org.readthedocs.build/en/99/

And there's an error in the console:

```
sidebar.js:34 Uncaught TypeError: Cannot read properties of null (reading 'querySelector')
    at HTMLDocument.initialiseSidebar (sidebar.js:34:38)
```

From the last line of:

```js
  const bodyWrapper = document.getElementsByClassName("bodywrapper")[0]
  const sidebar = document.getElementsByClassName("sphinxsidebar")[0]
  const sidebarWrapper = document.getElementsByClassName('sphinxsidebarwrapper')[0]
  const sidebarButton = document.getElementById("sidebarbutton")
  const sidebarArrow = sidebarButton.querySelector('span')
```

**Comment by AA-Turner** [comment]
Updated the JavaScript, the previous version was attempting to access an element that had yet to be created. Thank you for the review!

A

**Review by hugovk — APPROVED** [review]
Working fine now, thanks!

**Comment by JulienPalard** [comment]
I just built https://docs.python.org/3.12/ after merging this PR.

**Comment by JulienPalard** [comment]
https://github.com/python/python-docs-theme/pull/126/files
